install_config/oab_broker_configuration.adoc

```diff
@@ -0,0 +1,732 @@
+|N
+
+|`images`
+|The list of images to be used with an OpenShift Container Registry.
+|N
+|===
+
+[[oab-config-registry-prod-dev]]
+=== Production or Development
+
+A _production_ broker configuration is designed to be pointed at a trusted
+container distribution registry, such as the Red Hat Container Catalog (RHCC):
+
+[source,yaml]
+----
+registry:
+  - name: rhcc
+    type: rhcc
+    url: https://registry.access.redhat.com
+    tag: v3.7
+    white_list: 
+      - ".*-apb$"
+  - type: local_openshift
+    name: localregistry
+    namespaces:
+      - openshift
+    white_list: []
+----
+
+However, a _development_ broker configuration is primarily used by developers
+working on the broker. To enable developer settings, set the registry name to
+`dev` and the `dev_broker` field in the `broker` section to `true`:
+
+[source,yaml]
+----
+registry:
+  name: dev
+----
+
+[source,yaml]
+----
+broker:
+  dev_broker: true
+----
+
+[[oab-config-registry-storing-creds]]
+=== Storing Registry Credentials
+
+The `auth_type` field in the `registry` section of the broker configuration
+determines how the broker should read the registry credentials, either the
+`config`, `secret`, or `file` type.
+
+With the `config` type, the registry credentials are read from the broker
+configuration's `user` and `pass` values in the `registry` section:
+
+----
+registry:
+  - name: openshift
+    type: openshift
+    auth_type: config
+    user: <user>
+    pass: <password>
+...
+----
+
+If you want to ensure these credentials are not publicly accessible, you can use
+the `secret` type to configure a registry to use a secret from the broker's
+namespace. Alternatively, you can use the `file` to configure a registry to use
+a secret mounted as a volume.
+
+To use the `secret` or `file` type:
+
+. The associated secret should have the values `username` and `password` defined.
+When using a secret, you must ensure that the `openshift-ansible-service-broker`
+namespace exists, as this is where the secret will be read from.
++
+For example, create a *_reg-creds.yaml_* file:
++
+----
+$ cat reg-creds.yaml
+---
+username: <username>
+password: <password>
+----
+
+. Create a secret from this file in the `openshift-ansible-service-broker`
+namespace:
++
+----
+$ oc create secret generic \
+    registry-credentials-secret \
+    --from-file reg-creds.yaml \
+    -n openshift-ansible-service-broker
+----
+
+. Choose whether you want to use the `secret` or `file` type:
++
+--
+- To use the `secret` type, in the broker configuration, set `auth_type` to
+`secret` and `auth_name` to the name of the secret:
++
+----
+registry:
+  - name: isv
+    type: openshift
+    url: registry.connect.redhat.com
+    auth_type: secret
+    auth_name: registry-credentials-secret
+----
+
+- To use the `file` type:
+
+.. Edit the `asb` deployment configuration to mount your file into
+*_/tmp/registry-credentials/reg-creds.yaml_*:
++
+----
+$ oc edit asb -n openshift-ansible-service-broker
```

Comment:
This command should be `oc edit dc/asb -n openshift-ansible-service-broker`.

Reply:
Whoops! Thanks.